Python/errors.c

```diff
@@ -1697,13 +1697,17 @@ err_programtext(PyThreadState *tstate, FILE *fp, int lineno)
 PyObject *
 PyErr_ProgramText(const char *filename, int lineno)
 {
-    FILE *fp;
-    if (filename == NULL || *filename == '\0' || lineno <= 0) {
+    if (filename == NULL) {
         return NULL;
     }
-    PyThreadState *tstate = _PyThreadState_GET();
-    fp = _Py_fopen(filename, "r" PY_STDIOTEXTMODE);
-    return err_programtext(tstate, fp, lineno);
+
+    PyObject *filename_obj = PyUnicode_DecodeFSDefault(filename);
+    if (filename_obj == NULL) {
+        return NULL;
```

Comment:
Are you sure that the contract of this function allows raising an exception? It seems that `PyErr_ProgramTextObject` takes care to clear the exception, as does `err_programtext`.

Reply:
I don't know. In case of doubt, I added a PyErr_Clear() call.

My first intent was to deprecate the function. IMO it was exposed by mistake in the stable ABI. But since PyErr_ProgramTextObject() is still needed, I prefer to rewrite PyErr_ProgramText() with PyErr_ProgramTextObject().

Reply:
Thanks. Better safe than sorry :)